include/deal.II/base/multithread_info.h

```diff
@@ -117,23 +117,10 @@ class MultithreadInfo
 
 private:
   /**
-   * Private function to determine the number of CPUs. Implementation for
-   * Linux, OSF, SGI, and Sun machines; if no detection of the number of CPUs
-   * is supported, or if detection fails, this function returns one.
+   * Private function returning the maximum number of threads.
    */
-  static unsigned int
-  get_n_cpus();
-
-  /**
-   * Variable representing the maximum number of threads.
-   */
-  static unsigned int n_max_threads;
-
-  /**
-   * Variable representing the number of cores in the system. This is computed
-   * by get_n_cpus() and is returned by n_cores().
-   */
-  static const unsigned int n_cpus;
+  static unsigned int &
+  get_n_max_threads();
```

Comment:
Why is this necessary? Static initialization of `unsigned int` variables is pretty uncritical (even if it involves a function call) because it happens before `main()` even starts.

Reply:
Well, all static initialization of global variables happens before `main`, but the initialization order is undefined. The whole PR is about this at the end whether it probable to be a problem or not.

I guess you are right that it is really hard to run into an error here though...